identify/README.md

```diff
@@ -0,0 +1,56 @@
+# Identify v1.0.0
+
+The identify protocol is used to identify a peer and its capabilities.
+
+The protocol works by opening a stream, using `/ipfs/id/1.0.0` as the protocol string.
+The peer being identified responds by returning an `Identify` message and closes the
+stream.
+
+## The Identify Message
+
+```protobuf
+message Identify {
+  optional string protocolVersion = 5;
+  optional string agentVersion = 6;
+  optional bytes publicKey = 1;
+  repeated bytes listenAddrs = 2;
+  optional bytes observedAddr = 4;
+  repeated string protocols = 3;
+}
+```
+
+### protocolVersion
+
+The protocol version identifies the family of protocols used by the peer.
+The current protocol version is `ipfs/0.1.0`; if the protocol does not match
+the protocol used by the initiating peer, then the connection is considered
+unusable and the peer must close the connection.
+
+### agentVersion
+
+This is a free-form string, identifying the implementation of the peer.
```

Comment:
can we encourage a particular format, _even if non-normative_? the default is `go-libp2p/<semver>`, and IPFS changes this to `"go-ipfs/" + version.CurrentVersionNumber + "/" + version.CurrentCommit`.

https://github.com/ipfs/go-ipfs/blob/master/core/core.go#L99

Reply:
nah, let's let people call their agents whatever they want!

Reply:
although an example might be useful.

Reply:
example sounds good